testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/javascript/JavascriptAdapterTest.java

```diff
@@ -400,6 +397,69 @@ public void loginRequiredAction() {
                 .init(defaultArguments(), this::assertSuccessfullyLoggedIn);
     }
 
+    /**
+     * Test for scope handling via explicit Keycloak configuration: <pre>{@code
+     * new Keycloak({.... scope: "profile email phone"})
+     * }</pre>
+     * See KEYCLOAK-14412
+     */
+    @Test
+    public void testOnLoadRequireLoginShouldHonorScopes() {
+
+        JSObjectBuilder config = JSObjectBuilder.create()
+                .add("url", authServerContextRootPage + "/auth")
+                .add("realm", REALM_NAME)
+                .add("clientId", CLIENT_ID)
+                .add("scope", "profile email phone") // phone is optional client scope
+                ;
+
+        JSObjectBuilder initOptions = defaultArguments().loginRequiredOnLoad();
+
+        try {
+            testExecutor.configure(config)
+                        .init(initOptions);
+            // This throws exception because when JavascriptExecutor waits for AsyncScript to finish
+            // it is redirected to login page and executor gets no response
+
+            throw new RuntimeException("Probably the login-required OnLoad mode doesn't work, because testExecutor should fail with error that page was redirected.");
+        } catch (WebDriverException ex) {
+            // should happen
+        }
+
+        testExecutor.loginForm(testUser, this::assertOnTestAppUrl)
```

Comment:
If I am not mistaken, before this line, you can do also something like that.

```java
assertThat(jsDriver.getCurrentUrl(), constainsString("&scope=profile email phone"))
```

Reply:
I gave this a try:
```java
assertThat(jsDriver.getCurrentUrl(), containsString("&scope=openid%20profile%20email%20phone"));
```
seems to work.